yacc/reader.c

```diff
@@ -261,9 +261,29 @@ void process_apostrophe_body(FILE *f)
 
 
 static void process_open_curly_bracket(FILE *f) {
-    if (In_bitmap(caml_ident_start, *cptr) || *cptr == '|')
+    char *idcptr = cptr;
+
+    if (*idcptr == '%') {
+        if (*++idcptr == '%') idcptr++;
+
+        if (In_bitmap(caml_ident_start, *idcptr)) {
+            idcptr++;
+            while (In_bitmap(caml_ident_body, *idcptr)) idcptr++;
+            while (*idcptr == '.') {
```

Comment:
There is a missing ++idcptr here that may trigger an infinite loop o, `{%foo.bar` .

Reply:
Fixed and added to the tests.